ViewControllers/StockFeedViewController.m

```diff
@@ -88,6 +88,14 @@ - (void)searchBar:(UISearchBar *)searchBar textDidChange:(NSString *)searchText{
             if(resultsRange.location != NSNotFound){
                 [self.filteredStocksArray addObject:stock];
             }
+            NSRange companyNameResultsRange = [stock.companyName rangeOfString:searchText options:NSCaseInsensitiveSearch];
+            if(companyNameResultsRange.location != NSNotFound){
+                [self.filteredStocksArray addObject:stock];
+            }
+            NSRange exchangeResultsRange = [stock.exchange rangeOfString:searchText options:NSCaseInsensitiveSearch];
+            if(exchangeResultsRange.location != NSNotFound){
+                [self.filteredStocksArray addObject:stock];
+            }
```

Comment:
This might be why you are seeing duplicate stocks sometimes, I believe what you want to do is combine this into a single `if`